sycl/test-e2e/GroupAlgorithm/group_load_store.cpp

```diff
@@ -0,0 +1,213 @@
+// Use per-kernel split as a workaroud for a miscompilation bug in IGC.
+// RUN: %clangxx -fsycl -fsycl-targets=%{sycl_triple} -fsycl-device-code-split=per_kernel  %s -o %t.out
+// RUN: %{run} %t.out
+
+#include <sycl/sycl.hpp>
+
+#include <algorithm>
+#include <iomanip>
+
+using namespace sycl;
+namespace sycl_ext = sycl::ext::oneapi::experimental;
+
+constexpr int SG_SIZE = 16;
+constexpr int N_WGS = 3;
+
+template <typename MemType, typename SpanType, int ELEMS_PER_WI,
+
+          sycl_ext::group_algorithm_data_placement data_placement,
+          memory_scope scope>
+void test(size_t wg_size) {
+  std::cout << __PRETTY_FUNCTION__ << std::endl;
+
+  size_t global_size = wg_size * N_WGS;
+  queue q;
+
+  buffer<MemType, 1> b(global_size * ELEMS_PER_WI);
+  buffer<bool, 1> result(global_size);
+
+  q.submit([&](handler &cgh) {
+    accessor acc{b, cgh};
+    accessor res{result, cgh};
+
+    static constexpr auto scratch_mem_per_wi = ELEMS_PER_WI * sizeof(MemType);
```

Comment:
will we implement ```template<typename T, std::size_t ElementsPerWorkItem>
constexpr std::size_t memory_required(sycl::memory_scope scope,
                                      std::size_t block_size);``` function or is too uncertain how to use it? (I assume such function might appear in other proposals as well) https://github.com/intel/llvm/blob/ec562fcc2d6bffd8d2396b2c9f916698e06f2e2d/sycl/doc/extensions/proposed/sycl_ext_oneapi_group_load_store_extended.asciidoc?plain=1#L190